### PR TITLE
DirectusNext - Imgix respects responsive image sizes

### PR DESCRIPTION
**Pull request by poclerson**
## Issue
Closes https://github.com/OKAMca/bellmedia-mono/issues/1315

## Implementation details
`imgixImageLoader` set imgix `h` from the intrinsic Directus `height` while `w` is the responsive per-srcset width. Since `next/image` calls the loader once per srcset candidate, `fit=crop` paired an intrinsic `h` with a changing `w`, distorting the crop at every width but one.

- **Crop aspect ratio:** drop the fixed intrinsic `h`; for `fit=cover`, emit `ar` from the intrinsic `width:height` so imgix derives the crop height from the responsive `w`. Non-crop fits (`max`/`fill`/`min`) resolve from `w` alone.
- **Focal points:** normalise `fp-x`/`fp-y` against the intrinsic width/height (not the responsive render width, which drifted per srcset entry and could exceed imgix's `0..1` range); allow `0` as a valid top/left edge.
- **Refactor:** split `getDirectusFile` into `getDirectusImg`/`getDirectusVideo` wrappers so `getDirectusUrl`/`getDirectusFile` only map concerns common to every file type; image-only params (focal point, intrinsic dims, `fit`) now live in `getDirectusImg`. Added a `searchParams` prop to `TDirectusFileProps` for passing extra Directus-format params (e.g. `withoutEnlargement`).

> Note: `getDirectusFile` appends params in Directus format; mapping them to imgix is the loader's job. `ar` matches a true cover-crop only when the display box shares the intrinsic ratio — a different box ratio is a CSS concern (`sizes`) the loader can't infer.

## PR Checklist      
- [x] Lint must pass
- [x] Build must pass
- [x] There should be no warnings/errors in the console/terminal (check locally)

## How to test this PR
- Set NEXT_PUBLIC_IMGIX_SUBDOMAIN to "bmd-dev"
- Run the demo and open `/directus-next/imgix-image-loader/ecc3fb40-2aec-4d12-aec8-f580a76014a9` — a test bench covering every loader feature (fit modes, focal-point positions, entropy fallback, quality, `withoutEnlargement`, and fill + single/multi-viewport sizing).
- Inspect the generated `*.imgix.net` requests: `fit=crop` should carry `ar=<w>:<h>` (no fixed `h`), and `fp-x`/`fp-y` should stay stable across srcset candidates.

## Summary by CodeRabbit

- **New Features**
  - Added support for advanced Directus image transformations, including fit modes, focal-point cropping, quality settings, responsive sizing, and preventing image enlargement.
  - Improved image URL generation for more consistent results across responsive layouts and image loaders.
  - Added reusable utilities for generating Directus image and video properties.
  - Open Graph images now use the same image transformation handling.

- **Bug Fixes**
  - Improved focal-point cropping and aspect-ratio handling for responsive images.
  - Preserved custom image transformation parameters when generating asset URLs.

### CODE REVIEW

**Review by devAtOkam — COMMENTED** [review]
Summary: Fixes imgix responsive crop/focal math; safe if the `getDirectusFile` API change is intentional.

Task check: partially — the `DirectusImg` path fulfills OKAMca/bellmedia-mono#1315, but public utility callers can regress.

Issues:
- 🟡 `libs/directus/directus-next-component/src/utils/getDirectusFile/index.ts:68` — `getDirectusFile` is public and no longer appends width/height/focal params, so existing image callers using it with imgix lose `ar`/focal behavior unless they migrate to `getDirectusImg`.
- 💡 Keep image params in `getDirectusFile` for compatibility, or document this as a breaking migration.

**Comment by poclerson** [comment]
> Summary: Fixes imgix responsive crop/focal math; safe if the `getDirectusFile` API change is intentional.
> 
> Task check: partially — the `DirectusImg` path fulfills [OKAMca/bellmedia-mono#1315](https://github.com/OKAMca/bellmedia-mono/issues/1315), but public utility callers can regress.
> 
> Issues:
> 
> * 🟡 `libs/directus/directus-next-component/src/utils/getDirectusFile/index.ts:68` — `getDirectusFile` is public and no longer appends width/height/focal params, so existing image callers using it with imgix lose `ar`/focal behavior unless they migrate to `getDirectusImg`.
> * 💡 Keep image params in `getDirectusFile` for compatibility, or document this as a breaking migration.

That's intended. Documentation not needed since the last version adding imgix support was not used in any project

**Review by devAtOkam — APPROVED** [review]
✅ LGTM, ship it.

File: libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts [hunk 1]
```diff
@@ -42,38 +42,53 @@ function applyFitParams(params: URLSearchParams, directusParams: URLSearchParams
 
   const fpXPx = Number(directusParams.get('focal_point_x'))
   const fpYPx = Number(directusParams.get('focal_point_y'))
```
Comment (devAtOkam) [inline libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts]: Missing focal-point params are converted with `Number(null)`, which yields `0`. That makes an ordinary `fit=cover` image pass `hasValidFocalPoint` and emit `crop=focalpoint&fp-x=0&fp-y=0`, pinning the crop to the top-left instead of using entropy. Please check the raw params before coercion (while preserving an explicit `0`) and add automated coverage for missing, zero-edge, and out-of-range focal coordinates.

**Review by devAtOkam — DISMISSED** [review]
The responsive aspect-ratio fix looks sound, but the missing focal-point coercion introduces a visible top-left crop regression. Please fix that case and lock the focal-point behavior down with automated tests.

File: apps/demo/app/directus-next/imgix-image-loader/[id]/page.tsx [hunk 2]
```diff
@@ -1,22 +1,171 @@
...
+        </Case>
+        <Case title="Focal center" note="crop=focalpoint, fp-x=0.5, fp-y=0.5">
+          <DirectusImg {...file} width={1000} height={1000} focal_point_x={500} focal_point_y={500} fit="cover" className="h-full w-full object-cover" />
+        </Case>
+        <Case title="Focal right" note="crop=focalpoint, fp-x=1, fp-y=0.5">
+          <DirectusImg {...file} width={1000} height={1000} focal_point_x={1000} focal_point_y={500} fit="cover" className="h-full w-full object-cover" />
+        </Case>
+        <Case title="No focal → entropy" note="crop=entropy (no focal-point data)">
+          <DirectusImg {...file} width={1000} height={1000} fit="cover" className="h-full w-full object-cover" />
+        </Case>
```
[the remark this answers is not in the document]
Comment (poclerson) [reply apps/demo/app/directus-next/imgix-image-loader/[id]/page.tsx:103]: Fixed the wording — the section 2 description no longer implies it derives from section 1's 1600×900 landscape asset. It's its own independent 1000×1000 square fixture, as you suggested as the alternative resolution. Also added cases for the zero-focal and out-of-bounds-focal edge cases while I was in there.

File: libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts [hunk 3]
```diff
@@ -42,38 +42,53 @@ function applyFitParams(params: URLSearchParams, directusParams: URLSearchParams
...
+  // Derive the crop aspect ratio from the intrinsic dimensions and let imgix
+  // compute the height from the responsive `w`. Setting a fixed intrinsic `h`
+  // alongside a per-srcset `w` would change the crop ratio at every width.
+  if (hasValidDimensions)
+    params.set('ar', `${intrinsicWidth}:${intrinsicHeight}`)
+
+  // `fp-x`/`fp-y` are 0..1 fractions of the source, so they divide by the
+  // intrinsic width/height — never the responsive render width. `0` is a valid
+  // top/left edge; only the dimensions must be strictly positive.
+  const hasValidFocalPoint = [fpXPx, fpYPx].every(value => Number.isFinite(value) && value >= 0)
```
[the remark this answers is not in the document]
Comment (poclerson) [reply libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts]: Fixed — hasValidFocalPoint now also requires fpXPx <= intrinsicWidth && fpYPx <= intrinsicHeight, falling back to entropy otherwise. Added a demo case (out-of-bounds focal point) to cover it.

File: libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts (hunk 1 above)
Comment (poclerson) [reply libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts]: Good catch — fixed by reading the raw focal_point_x/y params before Number() coercion, so a missing param is no longer indistinguishable from an explicit 0. Added a demo case for the explicit-zero edge case (plus the out-of-bounds case CodeRabbit flagged) to exercise both fixes visually. This lib doesn't have a working automated test harness yet (checked the existing __tests__ file elsewhere in the repo — it fails with "describe is not defined", no vitest globals/test target wired up), so I covered these with the manual test bench rather than unit tests. Happy to set up vitest for this lib as a follow-up if you'd like it in CI.

Comment (poclerson) [reply libs/directus/directus-next/src/loaders/imgixImageLoader/index.ts]: Tested by human, fixed

**Review by marie-maxime — APPROVED** [review]
